packages/flutter_adaptive_scaffold/lib/src/breakpoints.dart

```diff
@@ -129,3 +131,49 @@ abstract class Breakpoint {
   /// the screen such as MediaQuery.sizeOf(context).width.
   bool isActive(BuildContext context);
 }
+
+/// An extension on [BuildContext] that returns the active [Breakpoint] based on
+/// an available [SlotLayout] ancestor or the default breakpoints.
+extension BreakpointExtension on BuildContext {
```

Comment:
Our style guide is partial against `extension` methods https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-extension . Maybe we can accomplish this through a static method in the `Breakpoint` class?

```dart
  static Breakpoint activeBreakpointOf(BuildContext context) {
    final SlotLayout? slotLayout = context.findAncestorWidgetOfExactType<SlotLayout>();
    if (slotLayout != null) {
      for (final MapEntry<Breakpoint, SlotLayoutConfig?> config
          in slotLayout.config.entries) {
        if (config.key.isActive(context)) {
          return config.key;
        }
      }
    }

    final TargetPlatform platform = Theme.of(context).platform;

    if (Breakpoint.desktop.contains(platform)) {
      if (Breakpoints.extraLarge.isActive(this)) {
        return Breakpoints.extraLargeDesktop;
      } else if (Breakpoints.large.isActive(this)) {
        return Breakpoints.largeDesktop;
      } else if (Breakpoints.mediumLarge.isActive(this)) {
        return Breakpoints.mediumLargeDesktop;
      } else if (Breakpoints.medium.isActive(this)) {
        return Breakpoints.mediumDesktop;
      } else if (Breakpoints.small.isActive(this)) {
        return Breakpoints.smallDesktop;
      }
    } else if (Breakpoint.mobile.contains(platform)) {
      if (Breakpoints.extraLarge.isActive(this)) {
        return Breakpoints.extraLargeMobile;
      } else if (Breakpoints.large.isActive(this)) {
        return Breakpoints.largeMobile;
      } else if (Breakpoints.mediumLarge.isActive(this)) {
        return Breakpoints.mediumLargeMobile;
      } else if (Breakpoints.medium.isActive(this)) {
        return Breakpoints.mediumMobile;
      } else if (Breakpoints.small.isActive(this)) {
        return Breakpoints.smallMobile;
      }
    }
    return Breakpoints.standard;
  }
```